code/datums/components/hallucinations.dm

```diff
@@ -0,0 +1,537 @@
+				src.ttl = world.time + timeout SECONDS //reset timeout
+			src.sound_prob = sound_prob
+			return TRUE //no duplicate
+		else
+			return FALSE //create a new hallucination
+
+//#########################################################
+//                    RANDOM IMAGE
+//#########################################################
+
+/// Random image - hallucinate an image on a visible tile with prob per life tick
+/datum/component/hallucination/random_image
+	var/list/image_list
+	var/image_prob = 10
+	var/image_time = 20 SECONDS
+
+	Initialize(timeout=30, image_list=null, image_prob=10, image_time=20 SECONDS)
+		.=..()
+		if(. == COMPONENT_INCOMPATIBLE || length(image_list) == 0)
+			return .
+		src.image_list = image_list
+		src.image_prob = image_prob
+		src.image_time = image_time
+
+	do_mob_tick(mob, mult)
+		if(probmult(image_prob))
+			//pick a non dense turf in view
+			var/list/turf/potentials = list()
+			for(var/turf/T in view(parent_mob))
+				if(T.density == 0)
+					potentials += T
+			var/turf/halluc_loc = pick(potentials)
+			var/image/halluc = new /image()
+			var/image/copyfrom = pick(src.image_list)
+			halluc.appearance = copyfrom.appearance
+			halluc.loc = halluc_loc
+			parent_mob.client?.images += halluc
+			SPAWN(src.image_time SECONDS)
+				qdel(halluc)
+		. = ..()
+
+	CheckDupeComponent(timeout, image_list, image_prob, image_time)
+		if(image_list ~= src.image_list) //this is the same hallucination, just update timeout and prob, time
+			if(timeout == -1)
+				src.ttl = timeout
+			else if(src.ttl != -1)
+				src.ttl = world.time + timeout SECONDS //reset timeout
+			src.image_prob = image_prob
+			src.image_time = image_time
+			return TRUE //no duplicate
+		else
+			return FALSE //create a new hallucination
+
+//#########################################################
+//                    FAKE ATTACK
+//#########################################################
+
+/// Fake attack - hallucinate being attacked by something
+/datum/component/hallucination/fake_attack
+	var/list/image_list
+	var/list/name_list
+	var/attacker_prob = 10
+	var/max_attackers = 5
+	var/attacker_list = list()
+	Initialize(timeout=30, image_list=null, name_list=null, attacker_prob=10, max_attackers=5)
+		.=..()
+		if(. == COMPONENT_INCOMPATIBLE)
+			return .
+		src.image_list = image_list
+		src.name_list = name_list
+		src.attacker_prob = attacker_prob
+		src.max_attackers = max_attackers
+
+	do_mob_tick(mob, mult)
+		//I know it's kinda gross, but whatever
+		for(var/obj/fake_attacker/fakey in src.attacker_list)
+			if(fakey.disposed)
+				src.attacker_list -= fakey
+		if(length(attacker_list) > src.max_attackers)
+			return
+		if(probmult(attacker_prob))
+			var/obj/fake_attacker/F
+			var/image/halluc
+			if(isnull(image_list)) //if not specified, let's do a 50/50 of critters or humans
+				var/list/possible_clones = new/list()
+				for(var/mob/living/carbon/human/H in mobs)
+					if (H.stat || H.lying || H.dir == NORTH) continue
```

Comment:
I don't think this code originates here but why can't we copy north-facing people??

Reply:
No idea, I just moved this from where it was to here. I assume there was a reason, but it was not commented.

Reply:
ok I'm kicking that one down the road